rflx/model.py

```diff
@@ -753,6 +765,21 @@ def __init__(
         if structure or types:
             self._prove()
 
+    def copy(
+        self,
+        full_name: str = None,
+        structure: Sequence[Link] = None,
+        types: Mapping[Field, Type] = None,
+    ) -> "Message":
+        return Message(
+            full_name if full_name else self.full_name,
+            structure if structure else copy(self.structure),
+            types if types else copy(self.types),
+        )
```

Comment:
Would it make more sense to us this to implement `__copy__` instead so it can be used by pythons `copy`?

Reply:
This copy method allows to create a copy of a message with some changes (e.g., `message.copy(structure=structure, types=types)`) without knowing the concrete message type (`UnprovenMessage`, `Message`, `DerivedMessage`, ...). The built-in `__copy__` method only allows an exact copy, so that changes must be done after copying. In our case this is not possible, as `structure` and `types` are evaluated in the constructor.